provider/ovh/ovh.go

```diff
@@ -97,6 +119,9 @@ func NewOVHProvider(ctx context.Context, domainFilter endpoint.DomainFilter, end
 		domainFilter:   domainFilter,
 		apiRateLimiter: ratelimit.New(apiRateLimit),
 		DryRun:         dryRun,
+		cacheInstance:  cache.New(cache.NoExpiration, cache.NoExpiration),
```

Comment:
Does this imply that we never expire the cache? Is this acceptable? What if the SOA records change?

Reply:
We are retrieving from the cache the SOA value that were assigned when we last call the OVHcloud API to retrieve the records.

Then, this value from cache is checked against the actual value of the SOA from the DNS zone, using a DNS query.
If the value are matching, the zone in cache is still up-to-date, and we can return records from cache.
Otherwise, we are re-fetching them.

In that case, having no expiration is acceptable, because the SOA DNS query against the DNS servers make us sure that the data in cache is usable.